db/models.py

```diff
@@ -1,52 +1,138 @@
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, User
 from django.db import models
+from django.core.exceptions import ValidationError
+from django.contrib.auth import get_user_model
 
 
-class Genre(models.Model):
-    name = models.CharField(max_length=255, unique=True)
+class CustomUserManager(BaseUserManager):
+    def create_user(self,
+                    username: str,
+                    password: str,
+                    **extra_fields) -> User:
```

Comment:
The return type of `create_user` should be `User`, which is your custom user model, not the `User` imported from `django.contrib.auth.models`. Consider changing the return type to `User`.